exo-go/src/service_restarter/index.go

```diff
@@ -61,22 +59,21 @@ func isRemove(event fsnotify.Event) bool {
 	return event.Op&fsnotify.Remove == fsnotify.Remove
 }
 
-func (s *ServiceRestarter) restart(watcherErrChannel chan<- error) error {
+func (s *ServiceRestarter) restart(watcherErrChannel chan<- error) {
 	if err := s.watcher.Close(); err != nil {
-		return err
+		watcherErrChannel <- err
```

Comment:
You will still need to return after pushing an error into the channel as we don't want the function to continue executing.